demo/api/src/news/generated/news.resolver.ts

```diff
@@ -26,27 +25,26 @@ import { PaginatedNews } from "./dto/paginated-news";
 export class NewsResolver {
     constructor(
         private readonly entityManager: EntityManager,
-        @InjectRepository(News) private readonly repository: EntityRepository<News>,
         private readonly blocksTransformer: BlocksTransformerService,
     ) {}
 
     @Query(() => News)
     @AffectedEntity(News)
     async news(@Args("id", { type: () => ID }) id: string): Promise<News> {
-        const news = await this.repository.findOneOrFail(id);
+        const news = await this.entityManager.findOneOrFail(News, id);
         return news;
     }
 
     @Query(() => News, { nullable: true })
     async newsBySlug(@Args("slug") slug: string, @Args("scope", { type: () => NewsContentScope }) scope: NewsContentScope): Promise<News | null> {
-        const news = await this.repository.findOne({ slug, scope });
+        const news = await this.entityManager.findOne(News, { slug, scope });
 
         return news ?? null;
     }
 
     @Query(() => PaginatedNews)
     async newsList(@Args() { scope, search, filter, sort, offset, limit }: NewsListArgs, @Info() info: GraphQLResolveInfo): Promise<PaginatedNews> {
-        const where = gqlArgsToMikroOrmQuery({ search, filter }, this.repository);
+        const where = gqlArgsToMikroOrmQuery({ search, filter }, this.entityManager.getRepository<News>(News));
```

Comment:
Imho you should find a way how we can get rid of the repository here as well.

Reply:
Yes I know I would do this today in an extra merge request this is an incompatible change

Reply:
The function only uses the entity metadata, so it should be possible to pass the metadata instead of the repository.

Reply:
yes it should but this is an incompatible change for all scaffolded resolvers. Do you want this change here or in an extra PR?

Reply:
Extra PR is okay. We can also write an upgrade script for this.

Reply:
https://github.com/vivid-planet/comet/pull/3727